### PR TITLE
kpt live end-to-end test script

### PR DESCRIPTION
**Pull request by seans3**
* First pass at `kpt live ...` end-to-end test script.
* Test script has to be run manually.
* TODO: Factor out helper functions
* TODO: Factor out test cases
* Updates `go.mod` and `go.sum` to use latest cli-utils which has a bug fix.

Example Output:
```
$ ./e2e/live/end-to-end-test.sh 
kpt end-to-end test

Building kpt...
/tmp/ci-9izetRftCQ/kpt

Deleting cluster "kind" ...
Creating cluster "kind" ...
 ✓ Ensuring node image (kindest/node:v1.17.0) 🖼
 ✓ Preparing nodes 📦  
 ✓ Writing configuration 📜 
 ✓ Starting control-plane 🕹️ 
 ✓ Installing CNI 🔌 
 ✓ Installing StorageClass 💾 
Set kubectl context to "kind-kind"
You can now use your cluster with:

kubectl cluster-info --context kind-kind

Have a nice day! 👋


Waiting for default service account...
-
default service account created

Testing basic ConfigMap init
kpt live init e2e/live/testdata/test-1
..SUCCESS

Testing initial preview
kpt live preview e2e/live/testdata/test-1
......SUCCESS

Testing basic apply
kpt live apply e2e/live/testdata/test-1
........SUCCESS

Testing basic preview
kpt live preview e2e/live/testdata/test-2
............SUCCESS

Testing basic prune
kpt live apply e2e/live/testdata/test-2
.............SUCCESS

Testing basic destroy
kpt live destroy e2e/live/testdata/test-2
........SUCCESS

kpt live init e2e/live/testdata/migrate-1
..SUCCESS

Testing kpt live apply with ConfigMap inventory
kpt live apply e2e/live/testdata/migrate-1
.........SUCCESS

Testing migrate from ConfigMap to ResourceGroup inventory
kpt live migrate e2e/live/testdata/migrate-1
..........SUCCESS

Testing kpt live preview with ResourceGroup inventory
kpt live preview e2e/live/testdata/migrate-1
.........SUCCESS

Testing kpt live apply/prune with ResourceGroup inventory
kpt live apply e2e/live/testdata/migrate-2
...........SUCCESS

Testing kpt destroy with ResourceGroup inventory
kpt live destroy e2e/live/testdata/migrate-2
........SUCCESS

Testing kpt live init for Kptfile (ResourceGroup inventory)
kpt live init e2e/live/testdata/migrate-3
....SUCCESS

Testing kpt live migrate with missing inventory-template.yaml should fail
kpt live migrate e2e/live/testdata/migrate-3
.SUCCESS

Testing kpt live migrate with no objects in cluster
kpt live migrate e2e/live/testdata/migrate-3
......SUCCESS

Testing kpt apply ResourceGroup inventory
kpt live apply e2e/live/testdata/migrate-3
.......SUCCESS


Cleaning up cluster
Deleting cluster "kind" ...
FINISHED
```

### CODE REVIEW

File: e2e/live/end-to-end-test.sh [hunk 1]
```diff
@@ -0,0 +1,451 @@
...
+assertContains "0 resource(s) pruned, 0 skipped"
+wait 2
+# Validate resources in the cluster
+# ConfigMap inventory with four inventory items.
+assertCMInventory "test-namespace" "4"
+printResult
+
+# Test 4: kpt live preview of apply/prune
+echo "Testing basic preview"
+echo "kpt live preview e2e/live/testdata/test-2"
```
Comment (Liujingfang1) [inline e2e/live/end-to-end-test.sh:244]: Can you add a comment for "test-2"? It contains the new set of resources for the same package in test-1.

File: e2e/live/end-to-end-test.sh [hunk 2]
```diff
@@ -0,0 +1,451 @@
...
+assertContains "0 resource(s) pruned, 0 skipped (preview)"
+# Validate resources in the cluster
+assertPodExists "pod-a" "test-rg-namespace"
+assertPodExists "pod-b" "test-rg-namespace"
+assertPodExists "pod-c" "test-rg-namespace"
+printResult
+
+# Test 10: kpt live apply/prune with ResourceGroup inventory
+echo "Testing kpt live apply/prune with ResourceGroup inventory"
+echo "kpt live apply e2e/live/testdata/migrate-2"
```
Comment (Liujingfang1) [inline e2e/live/end-to-end-test.sh:365]: Similarly, can you add a comment for "migrate-2"?

**Comment by mortent** [comment]
Can we run the script as a check on PRs? Otherwise we probably will not notice any issues until we are running it just before a release. 
We could also consider running a nightly job that updates the dependencies in kpt for cli-utils, kyaml and cmd/config and then compiles it and then runs the tests. It won't block PRs, but it would give us a an early heads up if there are changes in the dependencies that will break kpt.

**Comment by seans3** [comment]
> Can we run the script as a check on PRs? Otherwise we probably will not notice any issues until we are running it just before a release.
> We could also consider running a nightly job that updates the dependencies in kpt for cli-utils, kyaml and cmd/config and then compiles it and then runs the tests. It won't block PRs, but it would give us a an early heads up if there are changes in the dependencies that will break kpt.

I think there are several ways we could run this script more regularly, and give us a better signal. But I think the first step is to submit it so others can run it and see it working.

**Comment by mortent** [comment]
I think this looks good. If we end up adding more e2e-tests, I think we should consider doing this in go instead of bash.

**Review by mortent — APPROVED** [review]

**Pull request merged**